### PR TITLE
Test: add unit tests for BarChart component #1801

### LINKED ISSUE #1801: Add tests for <BarChart> component
Write unit tests for the `<BarChart>` React component to ensure expected behavior, edge case handling, and rendering logic.

## Essential Test Coverage Checklist

- [ ] **Renders successfully with minimal required props**  
- [ ] **Conditional rendering logic**  
- [ ] **Prop-based behavior** – different props affect output  
- [ ] **Event handling** – simulate user actions and verify callbacks  
- [ ] **State changes / internal logic**  
- [ ] **Default values and fallbacks**  
- [ ] **Text and content rendering**  
- [ ] **Handles edge cases and invalid inputs**  
- [ ] **Accessibility roles and labels**  
- [ ] **DOM structure / classNames / styles**

## Test Reference  
You can refer to the `AutoScrollToTop.test.tsx` file for an example of structure and best practices.  
To explore more examples, see the full component tests folder.

### PR DESCRIPTION
**Pull request by MohdWaqar98**
PR Description
Added comprehensive test cases for BarChart component

Tests cover rendering, data handling, and prop validation

Follows project testing standards

Proposed change

Resolves #1801 

This PR introduces unit tests for the BarChart component to ensure correctness across a range of scenarios. The tests validate prop-based behavior, rendering logic, edge case handling, and follow the structure defined in existing tests like AutoScrollToTop.test.tsx.

Checklist
 I've read and followed the [contributing guidelines](https://github.com/OWASP/Nest/blob/main/CONTRIBUTING.md).

 I've run make check-test locally; all checks and tests passed.

### CODE REVIEW

**Comment by kasya** [comment]
@MohdWaqar98 Thanks for working on this.

There are issues with your code that you need to address.
1. Run `make check` locally and address all issues that it shows (imports order, console in the code). Push changed files when it's all green and no error showing:
Screenshot 2025-07-29 at 5 36 29 PM

2. Update test cases to not have any `expect` in `if` statements - this is why we have unit tests. One check for each tests, no `if/else`.
3. Remove emojis from the tests and comments 👀 This is not acceptable for code.
4. Do not create a new PR for these changes - push everything to this same PR.

**Review by arkid15r — DISMISSED** [review]
Marking as `request changes`

File: frontend/__tests__/unit/components/BarChart.test.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+jest.mock('react-apexcharts', () => {
+  const MockChart = () => <div data-testid="mock-chart" />
+  return {
+    __esModule: true,
+    default: MockChart,
+  }
+})
+
+// Silence known warnings
+const originalError = console.error
```
Comment (kasya) [inline frontend/__tests__/unit/components/BarChart.test.tsx]: Any console statements are not allowed in tests.
We already have a mock Implementation for `console.error` in our `jest.setup.ts` so this is not needed here.

File: frontend/__tests__/unit/components/BarChart.test.tsx [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { faFire } from '@fortawesome/free-solid-svg-icons'
+import { act, render, screen } from '@testing-library/react'
+import '@testing-library/jest-dom'
+import { ThemeProvider } from 'next-themes'
+import React from 'react'
+
+import BarChart from 'components/BarChart'
+
+// Register FontAwesome icon
```
Comment (kasya) [inline frontend/__tests__/unit/components/BarChart.test.tsx:7]: I believe there are better ways to test this 🤔 Please check how other test files check for rendered icons.

**Review by kasya — CHANGES_REQUESTED** [review]
I fixed linting issues for you, but the test coverage is very low for this file.
Screenshot 2025-07-30 at 8 53 00 PM
47.36% coverage for statements, 0% for branches and 66.66% for functions. 
Could you at least get it to cover those lines mentioned on the right side?

**Comment by MohdWaqar98** [comment]
Hi @kasya ,

I've been working on this issue consistently for the past 4 days — going through your suggestions, fixing everything step by step, and writing as many test cases as I possibly could. I've managed to improve the coverage a bit, but I understand it's still below the expected threshold.

This is truly the best I could achieve with my current knowledge and resources. I really pushed myself because I wanted to contribute meaningfully, especially since I had a Level 1 issue merged earlier and was hopeful I could handle this one too.

If you feel the effort and partial improvements are worth accepting, I’d sincerely appreciate it. If not, I completely understand and respect your standards. Either way, I'm genuinely thankful for the opportunity — I’ve learned a lot through this process.

Thanks again,
Mohd Waqar

**Comment by kasya** [comment]
> Hi @kasya ,
> 
> If you feel the effort and partial improvements are worth accepting, I’d sincerely appreciate it. If not, I completely understand and respect your standards. Either way, I'm genuinely thankful for the opportunity — I’ve learned a lot through this process.
> 
> Thanks again, Mohd Waqar

Hi @MohdWaqar98! 
Thanks for being honest about it and not abandoning the PR. I'll fix some things that are failing right now (like multiple dynamic `expect` - this is not allowed) and then will merge in as is. We'll then have an extra issue to get this component's test coverage to higher numbers by someone else.

**Review by kasya — APPROVED** [review]
@MohdWaqar98 Thank you for working on this issue! I know it took time to figure out the test cases and even thought the coverage was low, I appreciate the enthusiasm.

I added more tests and it should be at 100% now for the component 👍🏼
